### PR TITLE
Feature/sof 4361 - ML units display their CSV file on the results tab during the predict phase

### PR DESCRIPTION
**Pull request by AcylSilane**

### CODE REVIEW

File: express/properties/workflow.py [hunk 1]
```diff
@@ -167,17 +167,28 @@ def _construct_predict_subworkflows(self, train_subworkflows: list) -> list:
...
                 # Set download-from-object-storage units
-                elif 'set-io-unit-filenames' in tags:
+                if 'set-io-unit-filenames' in tags:
                     self.set_io_unit_filenames(unit)
 
-                # Remove properties if needed
-                if "remove-all-results" in tags:
-                    unit["results"] = []
+                # Set predictors to print their predictions to the results tab during the training phase
+                if 'is_model' in tags:
```
Comment (timurbazhirov) [inline express/properties/workflow.py:186]: See https://github.com/Exabyte-io/exabyte-web-app/pull/1601/files#r610802899

Comment (AcylSilane) [reply express/properties/workflow.py:186]: Fixed

File: express/properties/workflow.py [hunk 2]
```diff
@@ -167,17 +167,28 @@ def _construct_predict_subworkflows(self, train_subworkflows: list) -> list:
...
                     self.set_io_unit_filenames(unit)
 
-                # Remove properties if needed
-                if "remove-all-results" in tags:
-                    unit["results"] = []
+                # Set predictors to print their predictions to the results tab during the predict phase
+                if 'creates-predictions-csv-during-predict-phase' in tags:
+                    unit["results"] = [{
+                        "name": "file_content",
+                        "basename": "predictions.csv",  # todo: We shouldn't be hardcoding this in to the flavors
```
Comment (triplepoint) [inline express/properties/workflow.py:189]: This should be ticketed up to address, if it needs to be addressed, or else it'll never get done.

**Review by triplepoint — APPROVED** [review]

File: express/properties/workflow.py (hunk 2 above)
Comment (AcylSilane) [reply express/properties/workflow.py:189]: Created - https://exabyte.atlassian.net/browse/SOF-4403

**Pull request merged**